src/rust/demikernel/bindings.rs

```diff
@@ -82,6 +82,15 @@ pub extern "C" fn demi_init(argc: c_int, argv: *mut *mut c_char) -> c_int {
         Err(e) => panic!("{:?}", e),
     };
 
+    // Check if demikernel has already been initialized and return
+    match unsafe { DEMIKERNEL.try_borrow() } {
+        Ok(libos) => match *libos {
+            Some(_) => return 0,
```

Comment:
Instead of continuing silently, how about returning `-EEXIST`?

Reply:
Sounds good. Updated.